extensions/internal-dictionary/src/index.tsx

```diff
@@ -0,0 +1,307 @@
+  handleDelete,
+  bulkActions,
+}: {
+  searchResult: DictionaryEntry;
+  handleCreate: (entry: DictionaryEntry) => void;
+  handleEdit: (entry: DictionaryEntry) => void;
+  handleDelete: (entry: DictionaryEntry) => void;
+  bulkActions: ReactNode;
+}) {
+  return (
+    <List.Item
+      title={searchResult.title}
+      subtitle={searchResult.description}
+      accessories={searchResult.score ? [{ text: (1 - searchResult.score).toFixed(2) }] : []}
+      actions={
+        <ActionPanel title="Actions">
+          <CreateDictionaryEntryAction onCreate={handleCreate} />
+          <EditDictionaryEntryAction onEdit={handleEdit} toEdit={{ ...searchResult, score: undefined }} />
+          <DeleteDictionaryEntryAction onDelete={handleDelete} toDelete={searchResult} />
+          <Action
+            title="Copy Description"
+            shortcut={{ modifiers: ["cmd"], key: "y" }}
+            icon={Icon.Clipboard}
+            onAction={() => {
+              Clipboard.copy(searchResult.description);
+              showToast({ title: "Copied Description" });
+            }}
+          />
+          {bulkActions}
+        </ActionPanel>
+      }
+    />
+  );
+}
+
+function UpdateEntryForm(props: { toEdit: DictionaryEntry; onEdit: (entry: DictionaryEntry) => void }) {
+  const { pop } = useNavigation();
+
+  function handleSubmit(values: { title: string; description: string }) {
+    props.onEdit({ title: values.title, description: values.description, id: props.toEdit.id });
+    pop();
+  }
+
+  const [newDesc, setNewDesc] = useState(props.toEdit.description);
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            title="Update Entry"
+            onSubmit={() => handleSubmit({ title: props.toEdit.title, description: newDesc })}
+          />
+        </ActionPanel>
+      }
+    >
+      <Form.Description title="Edit the description" text="To edit title, delete the entry and create a new one." />
+      <Form.TextField id="description" title="Description" value={newDesc} onChange={setNewDesc} />
+    </Form>
+  );
+}
+
+function CreateEntryForm(props: { onCreate: (entry: DictionaryEntry) => void }) {
+  const { pop } = useNavigation();
+
+  function handleSubmit(values: { title: string; description: string }) {
+    props.onCreate({ title: values.title, description: values.description, id: createId() });
+    pop();
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Create Entry" onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField id="title" title="Title" />
+      <Form.TextField id="description" title="Description" />
+    </Form>
+  );
+}
+
+function DeleteDictionaryEntryAction(props: { toDelete: DictionaryEntry; onDelete: (entry: DictionaryEntry) => void }) {
+  return (
+    <Action
+      icon={Icon.Trash}
+      shortcut={{ modifiers: ["cmd"], key: "d" }}
+      title="Delete Entry"
```

Comment:
You can set it to destructive: https://developers.raycast.com/api-reference/user-interface/actions#action.style